File: backend/cp25ssi2/src/main/java/sit/chat2date/cp25ssi2/dto/PlaceDTO.java
package sit.chat2date.cp25ssi2.dto;

import lombok.Data;

@Data
public class PlaceDTO {
    private String googlePlaceId;
    private String name;
    private double latitude;
    private double longitude;
    private String address;
    private String imageUrl;

    public PlaceDTO(String id, String name, double lat, double lng, String address, String imageUrl) {
        this.googlePlaceId = id;
        this.name = name;
        this.latitude = lat;
        this.longitude = lng;
        this.address = address;
        this.imageUrl = imageUrl;
    }
}

File: frontend-mobile/chat2date/lib/models/dto/place_dto.dart
class PlaceDTO {
  final String name;
  final String address;
  final String imageUrl;
  final String googlePlaceId;
  final double latitude;
  final double longitude;

  PlaceDTO({
    required this.name,
    required this.address,
    required this.imageUrl,
    required this.googlePlaceId,
    required this.latitude,
    required this.longitude,
  });

  factory PlaceDTO.fromJson(Map<String, dynamic> json) {
    return PlaceDTO(
      name: json['name'] ?? '',
      address: json['address'] ?? '',
      imageUrl: json['imageUrl'] ?? '',
      googlePlaceId: json['googlePlaceId'] ?? '',
      latitude: (json['latitude'] as num).toDouble(),
      longitude: (json['longitude'] as num).toDouble(),
    );
  }

  // Helper สำหรับแปลงไปใช้กับวงล้อ SpinWheel
  Map<String, dynamic> toWheelPrize() {
    return {
      "name": name,
      "imageUrl": imageUrl,
      "address": address,
      "lat": latitude,
      "lng": longitude,
    };
  }
}